lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -210,24 +248,47 @@ def _user_requested_to_enable(self, plugin: OpenSearchPlugin) -> bool:
     def _is_enabled(self, plugin: OpenSearchPlugin) -> bool:
         """Returns true if plugin is enabled."""
         # If not requested to be disabled, check if options are configured or not
-        are_configs_enabled = plugin.config().config_entries_to_add
-        stored_plugin_conf = self._opensearch_config.get_plugin(are_configs_enabled)
-        for key, val in stored_plugin_conf.items():
-            if are_configs_enabled.get(key) != val:
+        try:
+            are_configs_enabled = plugin.config().config_entries_to_add
+            stored_plugin_conf = self._opensearch_config.get_plugin(are_configs_enabled)
+            # Compare dicts
+            list_a = list(are_configs_enabled.keys())
+            list_a.sort()
+            list_b = list(stored_plugin_conf.keys())
+            list_b.sort()
+            if list_a != list_b:
+                # Keys do not match
                 return False
```

Comment:
Please expressive variable names and no single letters ones :) 

```suggestion
            plugin_conf = plugin.config().config_entries_to_add
            stored_plugin_conf = self._opensearch_config.get_plugin(plugin_conf)
            return plugin_conf == stored_plugin_conf
```